<h1> Palmer Penguin Data Analytics</h1>
<hr>
<h3>Analyst: Mary Angelie Remojo</h3>

In [42]:
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings('ignore')

<p> Importing dataset to Jupyter notebook </p>
<p>Converting dataset to dataframe</p>
<ul>Requirements
    <li>Name of the Dataframe</li>
    <li>Source</li>
</ul>

In [3]:
penguins = pd.read_csv(r'd:\ELEC02\penguins.csv')
penguins

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0       Adelie  Torgersen            39.1           18.7              181.0   
1       Adelie  Torgersen            39.5           17.4              186.0   
2       Adelie  Torgersen            40.3           18.0              195.0   
3       Adelie  Torgersen             NaN            NaN                NaN   
4       Adelie  Torgersen            36.7           19.3              193.0   
..         ...        ...             ...            ...                ...   
339  Chinstrap      Dream            55.8           19.8              207.0   
340  Chinstrap      Dream            43.5           18.1              202.0   
341  Chinstrap      Dream            49.6           18.2              193.0   
342  Chinstrap      Dream            50.8           19.0              210.0   
343  Chinstrap      Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[344 rows x 8 columns]

<h1> Descriptive Analytics</h1>

In [4]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      342.000000     342.000000         342.000000   342.000000   
mean        43.921930      17.151170         200.915205  4201.754386   
std          5.459584       1.974793          14.061714   801.954536   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.225000      15.600000         190.000000  3550.000000   
50%         44.450000      17.300000         197.000000  4050.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

In [5]:
penguins.describe().T

count         mean         std     min       25%      50%  \
bill_length_mm     342.0    43.921930    5.459584    32.1    39.225    44.45   
bill_depth_mm      342.0    17.151170    1.974793    13.1    15.600    17.30   
flipper_length_mm  342.0   200.915205   14.061714   172.0   190.000   197.00   
body_mass_g        342.0  4201.754386  801.954536  2700.0  3550.000  4050.00   
year               344.0  2008.029070    0.818356  2007.0  2007.000  2008.00   

                      75%     max  
bill_length_mm       48.5    59.6  
bill_depth_mm        18.7    21.5  
flipper_length_mm   213.0   231.0  
body_mass_g        4750.0  6300.0  
year               2009.0  2009.0

<h3>Cleaning the Data Frame</h3>

In [6]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h2>Selecting a column<h2>
<h3>Syntax</h3><br>
data_frame['column_name']

In [7]:
penguins[penguins['island'] == 'Dream']

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
30      Adelie  Dream            39.5           16.7              178.0   
31      Adelie  Dream            37.2           18.1              178.0   
32      Adelie  Dream            39.5           17.8              188.0   
33      Adelie  Dream            40.9           18.9              184.0   
34      Adelie  Dream            36.4           17.0              195.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
30        3250.0  female  2007  
31        3900.0    male  2007  
32        3300.0  female  2007  
33        3900.0    male  2007  
34        3325.0  female  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[124 rows x 8 columns]

In [8]:
penguins['species'].value_counts()

species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

In [9]:
penguins['species'].unique()

array(['Adelie', 'Gentoo', 'Chinstrap'], dtype=object)

In [10]:
chinstrap_df = penguins[penguins['species'] == "Chinstrap"]

In [11]:
chinstrap_df

species island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
276  Chinstrap  Dream            46.5           17.9              192.0   
277  Chinstrap  Dream            50.0           19.5              196.0   
278  Chinstrap  Dream            51.3           19.2              193.0   
279  Chinstrap  Dream            45.4           18.7              188.0   
280  Chinstrap  Dream            52.7           19.8              197.0   
..         ...    ...             ...            ...                ...   
339  Chinstrap  Dream            55.8           19.8              207.0   
340  Chinstrap  Dream            43.5           18.1              202.0   
341  Chinstrap  Dream            49.6           18.2              193.0   
342  Chinstrap  Dream            50.8           19.0              210.0   
343  Chinstrap  Dream            50.2           18.7              198.0   

     body_mass_g     sex  year  
276       3500.0  female  2007  
277       3900.0    male  2007  
278       3650.0    male  2007  
279       3525.0  female  2007  
280       3725.0    male  2007  
..           ...     ...   ...  
339       4000.0    male  2009  
340       3400.0  female  2009  
341       3775.0    male  2009  
342       4100.0    male  2009  
343       3775.0  female  2009  

[68 rows x 8 columns]

In [12]:
adelie_df = penguins[penguins['species'] == "Adelie"]

In [13]:
adelie_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen             NaN            NaN                NaN   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3            NaN     NaN  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [14]:
gentoo_df = penguins[penguins['species'] == "Gentoo"]

In [15]:
gentoo_df

species  island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
152  Gentoo  Biscoe            46.1           13.2              211.0   
153  Gentoo  Biscoe            50.0           16.3              230.0   
154  Gentoo  Biscoe            48.7           14.1              210.0   
155  Gentoo  Biscoe            50.0           15.2              218.0   
156  Gentoo  Biscoe            47.6           14.5              215.0   
..      ...     ...             ...            ...                ...   
271  Gentoo  Biscoe             NaN            NaN                NaN   
272  Gentoo  Biscoe            46.8           14.3              215.0   
273  Gentoo  Biscoe            50.4           15.7              222.0   
274  Gentoo  Biscoe            45.2           14.8              212.0   
275  Gentoo  Biscoe            49.9           16.1              213.0   

     body_mass_g     sex  year  
152       4500.0  female  2007  
153       5700.0    male  2007  
154       4450.0  female  2007  
155       5700.0    male  2007  
156       5400.0    male  2007  
..           ...     ...   ...  
271          NaN     NaN  2009  
272       4850.0  female  2009  
273       5750.0    male  2009  
274       5200.0  female  2009  
275       5400.0    male  2009  

[124 rows x 8 columns]

In [16]:
penguins[penguins['body_mass_g'] >= 4000]

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
7       Adelie  Torgersen            39.2           19.6              195.0   
9       Adelie  Torgersen            42.0           20.2              190.0   
14      Adelie  Torgersen            34.6           21.1              198.0   
17      Adelie  Torgersen            42.5           20.7              197.0   
19      Adelie  Torgersen            46.0           21.5              194.0   
..         ...        ...             ...            ...                ...   
323  Chinstrap      Dream            49.0           19.6              212.0   
329  Chinstrap      Dream            50.7           19.7              203.0   
333  Chinstrap      Dream            49.3           19.9              203.0   
339  Chinstrap      Dream            55.8           19.8              207.0   
342  Chinstrap      Dream            50.8           19.0              210.0   

     body_mass_g   sex  year  
7         4675.0  male  2007  
9         4250.0   NaN  2007  
14        4400.0  male  2007  
17        4500.0  male  2007  
19        4200.0  male  2007  
..           ...   ...   ...  
323       4300.0  male  2009  
329       4050.0  male  2009  
333       4050.0  male  2009  
339       4000.0  male  2009  
342       4100.0  male  2009  

[177 rows x 8 columns]

<h1>How many Chinstrap have a body mass that is around 4000g?</h1>

In [17]:
penguins[(penguins['species'] == "Chinstrap") &(penguins['body_mass_g'] >= 4000)].info()

<class 'pandas.core.frame.DataFrame'>
Index: 16 entries, 284 to 342
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   species            16 non-null     object 
 1   island             16 non-null     object 
 2   bill_length_mm     16 non-null     float64
 3   bill_depth_mm      16 non-null     float64
 4   flipper_length_mm  16 non-null     float64
 5   body_mass_g        16 non-null     float64
 6   sex                16 non-null     object 
 7   year               16 non-null     int64  
dtypes: float64(4), int64(1), object(3)
memory usage: 1.1+ KB


In [18]:
penguins.isnull().sum()

species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
year                  0
dtype: int64

<h1>FILLING UP ALL NULL VALUES</h1>

In [19]:
penguins['bill_length_mm'] = penguins['bill_length_mm'].fillna(penguins['bill_length_mm'].mode()[0])
penguins['bill_depth_mm'] = penguins['bill_depth_mm'].fillna(penguins['bill_depth_mm'].mode()[0])
penguins['flipper_length_mm'] = penguins['flipper_length_mm'].fillna(penguins['flipper_length_mm'].mode()[0])
penguins['body_mass_g'] = penguins['body_mass_g'].fillna(penguins['body_mass_g'].mode()[0])
penguins['sex'] = penguins['sex'].fillna(penguins['sex'].mode()[0])

In [20]:
penguins.isnull().sum()

species              0
island               0
bill_length_mm       0
bill_depth_mm        0
flipper_length_mm    0
body_mass_g          0
sex                  0
year                 0
dtype: int64

<h1>/(0o0)\</h1>

<h1>HOW TO CHANGE THE DATA TYPE OF A COLUMN</h1>

In [21]:
penguins['species'] = penguins['species'].astype('category')
penguins['island'] = penguins['island'].astype('category')
penguins['sex'] = penguins['sex'].astype('category')

In [22]:
penguins.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 344 entries, 0 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            344 non-null    category
 1   island             344 non-null    category
 2   bill_length_mm     344 non-null    float64 
 3   bill_depth_mm      344 non-null    float64 
 4   flipper_length_mm  344 non-null    float64 
 5   body_mass_g        344 non-null    float64 
 6   sex                344 non-null    category
 7   year               344 non-null    int64   
dtypes: category(3), float64(4), int64(1)
memory usage: 15.0 KB


In [23]:
penguins.describe()

bill_length_mm  bill_depth_mm  flipper_length_mm  body_mass_g  \
count      344.000000     344.000000         344.000000   344.000000   
mean        43.905523      17.150291         200.851744  4199.418605   
std          5.447882       1.969061          14.045266   800.197923   
min         32.100000      13.100000         172.000000  2700.000000   
25%         39.275000      15.600000         190.000000  3550.000000   
50%         44.250000      17.300000         197.000000  4025.000000   
75%         48.500000      18.700000         213.000000  4750.000000   
max         59.600000      21.500000         231.000000  6300.000000   

              year  
count   344.000000  
mean   2008.029070  
std       0.818356  
min    2007.000000  
25%    2007.000000  
50%    2008.000000  
75%    2009.000000  
max    2009.000000

<h1>EXPLORATORY DATA ANALYTICS (EDA)</h1>

<br>dummy questions</br>

<br>what species is the most abundant</br>
<br>What species have the oldest year</br>
<br>How many Adelie lives in Dream Island</br>
<br>what species have the most bill lenght</br>
<br>how many female Gentoo is in the Database</br>
<br>how many Male Gentoo is in the Database</br>
<br>what species have the most body mass around 3000kg</br>

<h1> 1. What species is the most abundant?</h1>

In [24]:
chinstrap_df = penguins[penguins['species'] == "Chinstrap"]
adelie_df = penguins[penguins['species'] == "Adelie"]
gentoo_df = penguins[penguins['species'] == "Gentoo"]

In [25]:
chinstrap_df['species'].info()

<class 'pandas.core.series.Series'>
Index: 68 entries, 276 to 343
Series name: species
Non-Null Count  Dtype   
--------------  -----   
68 non-null     category
dtypes: category(1)
memory usage: 744.0 bytes


In [26]:
adelie_df['species'].info()

<class 'pandas.core.series.Series'>
Index: 152 entries, 0 to 151
Series name: species
Non-Null Count  Dtype   
--------------  -----   
152 non-null    category
dtypes: category(1)
memory usage: 1.5 KB


In [27]:
gentoo_df['species'].info()

<class 'pandas.core.series.Series'>
Index: 124 entries, 152 to 275
Series name: species
Non-Null Count  Dtype   
--------------  -----   
124 non-null    category
dtypes: category(1)
memory usage: 1.2 KB


<h3>INSIGHT #1</h3>
<h4>The most abundant species is the Adelie specie.</h4>

<h1>2. What species have the oldest year?</h1>

In [28]:
chinstrap_df['year'].mean()

np.float64(2007.9705882352941)

In [29]:
adelie_df['year'].mean()

np.float64(2008.0131578947369)

In [30]:
gentoo_df['year'].mean()

np.float64(2008.0806451612902)

<h3>Insight #2</h3>
<h4>The species with the oldest year is the Gentoo and Adelie species.</h4>

<h1>3. How many Adelie specie lives in Dream Island?</h1>

In [31]:
adelie_df = penguins[penguins['species'] == "Adelie"]
adelie_df

species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0    Adelie  Torgersen            39.1           18.7              181.0   
1    Adelie  Torgersen            39.5           17.4              186.0   
2    Adelie  Torgersen            40.3           18.0              195.0   
3    Adelie  Torgersen            41.1           17.0              190.0   
4    Adelie  Torgersen            36.7           19.3              193.0   
..      ...        ...             ...            ...                ...   
147  Adelie      Dream            36.6           18.4              184.0   
148  Adelie      Dream            36.0           17.8              195.0   
149  Adelie      Dream            37.8           18.1              193.0   
150  Adelie      Dream            36.0           17.1              187.0   
151  Adelie      Dream            41.5           18.5              201.0   

     body_mass_g     sex  year  
0         3750.0    male  2007  
1         3800.0  female  2007  
2         3250.0  female  2007  
3         3800.0    male  2007  
4         3450.0  female  2007  
..           ...     ...   ...  
147       3475.0  female  2009  
148       3450.0  female  2009  
149       3750.0    male  2009  
150       3700.0  female  2009  
151       4000.0    male  2009  

[152 rows x 8 columns]

In [32]:
adelie_df['island'].info()

<class 'pandas.core.series.Series'>
Index: 152 entries, 0 to 151
Series name: island
Non-Null Count  Dtype   
--------------  -----   
152 non-null    category
dtypes: category(1)
memory usage: 1.5 KB


<h3>Insight #3</h3>
<h4>There are a total of 152 Adelie in the Dream Island.</h4>

<h1>4. What species have the most bill lenght?</h1>

In [33]:
chinstrap_df['bill_length_mm'].mean()

np.float64(48.83382352941177)

In [34]:
adelie_df['bill_length_mm'].mean()

np.float64(38.80657894736842)

In [35]:
gentoo_df['bill_length_mm'].mean()

np.float64(47.45322580645161)

<h3>Insight #4</h3>
<h4>The species with the most bill lenght is the Chinstrap specie.</h4>

<h1>5. How many female Gentoo is in the Database?</h1>

In [36]:
((penguins['species'] == 'Gentoo') & (penguins['sex'] == 'female')).sum()

np.int64(58)

<h1>INSIGHT #5</h1>
<h3>There is a total of 58 Female Gentoo specie.</h3>

<h1>6. How many Male Gentoo is in the Database?</h1>

In [37]:
((penguins['species'] == 'Gentoo') & (penguins['sex'] == 'male')).sum()

np.int64(66)

<h1>INSIGHT #6</h1>
<h3>There is a total of 66 Male Gentoo specie.</h3>

<h1>7. What species have the most body mass around 3000kg?</h1>

In [38]:
penguins[(penguins['species'] == "Chinstrap") &(penguins['body_mass_g'] >= 3000)].info()

<class 'pandas.core.frame.DataFrame'>
Index: 66 entries, 276 to 343
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            66 non-null     category
 1   island             66 non-null     category
 2   bill_length_mm     66 non-null     float64 
 3   bill_depth_mm      66 non-null     float64 
 4   flipper_length_mm  66 non-null     float64 
 5   body_mass_g        66 non-null     float64 
 6   sex                66 non-null     category
 7   year               66 non-null     int64   
dtypes: category(3), float64(4), int64(1)
memory usage: 3.7 KB


In [39]:
penguins[(penguins['species'] == "Adelie") &(penguins['body_mass_g'] >= 3000)].info()

<class 'pandas.core.frame.DataFrame'>
Index: 145 entries, 0 to 151
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            145 non-null    category
 1   island             145 non-null    category
 2   bill_length_mm     145 non-null    float64 
 3   bill_depth_mm      145 non-null    float64 
 4   flipper_length_mm  145 non-null    float64 
 5   body_mass_g        145 non-null    float64 
 6   sex                145 non-null    category
 7   year               145 non-null    int64   
dtypes: category(3), float64(4), int64(1)
memory usage: 7.6 KB


In [40]:
penguins[(penguins['species'] == "Gentoo") &(penguins['body_mass_g'] >= 3000)].info()

<class 'pandas.core.frame.DataFrame'>
Index: 124 entries, 152 to 275
Data columns (total 8 columns):
 #   Column             Non-Null Count  Dtype   
---  ------             --------------  -----   
 0   species            124 non-null    category
 1   island             124 non-null    category
 2   bill_length_mm     124 non-null    float64 
 3   bill_depth_mm      124 non-null    float64 
 4   flipper_length_mm  124 non-null    float64 
 5   body_mass_g        124 non-null    float64 
 6   sex                124 non-null    category
 7   year               124 non-null    int64   
dtypes: category(3), float64(4), int64(1)
memory usage: 6.6 KB


<h1>INSIGHT #7</h1>
<h3>Adelie specie have the most penguins with a body mass around 3000kg</h3>

<h1>DATA AGGREGATION</h1>

<h1>8. Which species has the highest average flipper length?</h1>

In [43]:
penguins.groupby('species')['flipper_length_mm'].mean()

species
Adelie       189.953947
Chinstrap    195.823529
Gentoo       216.967742
Name: flipper_length_mm, dtype: float64

<h1>INSIGHT #8</h1>
<h3>The Gentoo species has the highest average flipper length among the three species.</h3>

<h1>9. What male species have the highest body mass?</h1>

In [45]:
penguins[penguins['sex'] == 'male'].groupby('species')['body_mass_g'].mean()

species
Adelie       4008.544304
Chinstrap    3938.970588
Gentoo       5404.924242
Name: body_mass_g, dtype: float64

<h1>INSIGHT #9</h1>
<h3>Male Gentoo penguins have the highest body mass.</h3>

<h1>10. What female species have the highest body mass?</h1>

In [46]:
penguins[penguins['sex'] == 'female'].groupby('species')['body_mass_g'].mean()

species
Adelie       3368.835616
Chinstrap    3527.205882
Gentoo       4679.741379
Name: body_mass_g, dtype: float64

<h1>INSIGHT #10</h1>
<h3>Female Gentoo penguins also have the highest body mass.</h3>

<h1>11. Which island has the most penguins?</h1>

In [47]:
penguins['island'].value_counts()

island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

<h1>INSIGHT #11</h1>
<h3>Biscoe Island has the highest number of penguins in the dataset.</h3>

<h1>12. Which species has the smallest average bill depth?</h1>

In [48]:
penguins.groupby('species')['bill_depth_mm'].mean()

species
Adelie       18.337500
Chinstrap    18.420588
Gentoo       14.998387
Name: bill_depth_mm, dtype: float64

<h1>INSIGHT #12</h1>
<h3>The Gentoo species has the smallest average bill depth.</h3>

<h1>13. How many penguins have flipper length greater than 210mm per species?</h1>

In [49]:
penguins[penguins['flipper_length_mm'] > 210].groupby('species').size()

species
Adelie        0
Chinstrap     1
Gentoo       99
dtype: int64

<h1>INSIGHT #13</h1>
<h3>Most penguins with flipper length over 210mm is the Gentoo species.</h3>